src/main/scala/beam/router/model/BeamPath.scala

```diff
@@ -60,17 +60,18 @@ case class BeamPath(
     )
   }
 
+  @SuppressWarnings(Array("UnsafeTraversableMethods"))
   def linkAtTime(tick: Int): Int = {
     tick - startPoint.time match {
       case secondsAlongPath if secondsAlongPath <= 0 || linkIds.size <= 1 =>
+        // TODO: there is a likely bug here because linkIds.size can be 0(zero)
```

Comment:
Issue-ize? Although I don't see the issue per se

Reply:
@JustinPihony the following is the whole condition:
```Scala
case secondsAlongPath if secondsAlongPath <= 0 || linkIds.size <= 1 =>
  linkIds.head
```
if `linkIds.size == 0` then `linkIds.head` will throw an exception. That is why looks like a bug to me

Reply:
Sorry - I honestly didn't read the TODO itself - went into auto-pilot to ask a TODO to be issue-ized. That said, I looked at the blame and found [this](https://github.com/LBNL-UCB-STI/beam/commit/49899e973407383f50be779e1621aa3902197469#diff-161b4a7c3d656317ee6511bbf2212cfa249d2fb042e2e087b5af63d9de206f82). Upon review I am questioning a number of those changes. This definitely should be raised as a new issue as there are a lot of flaws - and should have tests to catch.